doc/developers-guide/Basic-iq-handler.md

```diff
@@ -0,0 +1,206 @@
+## Basic IQ Handler
+This tutorial will show you how to add & test a simple module with an IQ
+handler to MongooseIM.
+`gen_iq_handler` module provides functionality for registering IQ
+handlers for specific namespaces.
+
+
+### Clone & build
+
+See [How-to-build](../../user-guide/How-to-build) for details on building MongooseIM
+from source code.
+
+
+### Create module & add basic IQ handler
+
+* Go to `src` and create a basic module implementing `gen_mod` behaviour
+* In `start/2` register the IQ handler with specified namespace, type
+  (processing iq policy) and function which will handle the incoming IQ stanza
+* In `stop/1` remove the registered handler
+* Implement the function for handler:
+    * If the incoming IQ stanza is of type `get` or `set` it will be
+      returned with the type set to `result`
+    * If the server doesn't recognise the hostname, the returning stanza
+      will be of type `error`
+
+See [Server Rules for Processing XML Stanzas](https://tools.ietf.org/html/rfc6120#section-10) for more
+detailed information on the topic.
+
+```erlang
+-module(mod_iq_example).
+-behaviour(gen_mod).
+
+-include("mongoose.hrl").
+-include("jlib.hrl").
+
+%% gen_mod callbacks
+-export([start/2, stop/1]).
+
+%% IQ handlers
+-export([process_iq/4]).
+
+start(Host, _Opts) ->
+    gen_iq_handler:add_iq_handler(ejabberd_sm, Host, <<"erlang-solutions.com:example">>,
+                                  ?MODULE, process_iq, no_queue).
```

Comment:
Would it make sense to describe possible options here? I mean `no_queue`, `one_queue` etc. Unless they are already documented somewhere else or it's out of scope of this doc :)

Reply:
I haven't found such a description, so I think it would :)